### PR TITLE
Preserve execution context when running code in ThreadPool

### PR DESCRIPTION
**Pull request by RobbeSneyders**
Currently contextvars from the ASGI context are not preserved into the WSGI context, which can lead to issues with certain middleware.
- Eg. see https://github.com/django/asgiref/pull/73
- In [Connexion](https://github.com/spec-first/connexion), we want to [leverage contextvars to provide context to the application](https://github.com/spec-first/connexion/blob/feature/async-app/connexion/context.py)

### CODE REVIEW

File: a2wsgi/wsgi.py [hunk 1]
```diff
@@ -1,12 +1,19 @@
...
 import collections
+import functools
 import os
 import sys
 import typing
 from concurrent.futures import ThreadPoolExecutor
 
 from .types import Environ, Message, Receive, Scope, Send, StartResponse, WSGIApp
 
+try:
```
Comment (abersheeran) [inline a2wsgi/wsgi.py:11]: The new version of a2wsgi only supports above 3.7, so we can use contextvars directly.

**Comment by abersheeran** [comment]
Sorry for the lateness, I have been off for the last few days due to COVID-19.

File: a2wsgi/wsgi.py (hunk 1 above)
Comment (RobbeSneyders) [reply a2wsgi/wsgi.py]: Aha, didn't know that.
I checked the [pyproject.toml](https://github.com/abersheeran/a2wsgi/blob/master/pyproject.toml#L11) which still includes 3.6.

**Comment by RobbeSneyders** [comment]
No worries, was not expecting a response over Christmas anyway. Hope you feel better!

File: a2wsgi/wsgi.py (hunk 1 above)
Comment (abersheeran) [reply a2wsgi/wsgi.py]: Ah, that was an oversight on my part.

**Comment by RobbeSneyders** [comment]
@abersheeran could this be released? We'd like to remove the monkeypatch we have right now in [connexion](https://github.com/spec-first/connexion).

**Comment by abersheeran** [comment]
> @abersheeran could this be released? We'd like to remove the monkeypatch we have right now in [connexion](https://github.com/spec-first/connexion).

https://pypi.org/project/a2wsgi/1.7.0/ Thanks for your contribution, this PR is now released in 1.7.

**Pull request merged**